kodi_addon_checker/check_files.py

```diff
@@ -89,6 +89,14 @@ def addon_xml_matches_folder(report: Report, addon_path: str, parsed_xml, folder
             report.add(Record(PROBLEM, "Addon id and folder name does not match."))
 
 
+def check_for_legacy_changelog(report: Report, addon_path: str):
+    """Check whehter a legacy changelog.txt is present in the addon folder root"""
+    changelog_path = os.path.join(addon_path, "changelog.txt")
+    if os.path.isfile(changelog_path):
+        report.add(Record(PROBLEM, "A changelog.txt file was found. Use the <news> tag in addon.xml "
```

Comment:
I am not sure others will agree with the problem. @razzeee, @rechi what do you guys think?

Reply:
We can make it fail for Leia and Matrix, and warn for Krypton?

Reply:
That would be better. there are still a lot of addons in krypton using the old changelog, I'm not confortable with hardfailing the addon-checker because of that. Matrix and Leia don't have much addons and we've been enforcing it anyway.

Reply:
I'm actually using both. As news only should include the latest changes. But the changelog will even have older versions.

Reply:
@Razzeee the thing is the changelog file tends to grow when you create new releases and is a helper file that is not parsed by kodi anymore, so there's no reason to be shipping it from the mirrors and making addon packages bigger. Ideally you should be able to use anyfiles you want in your upstream repo but avoid submitting it to the repo (adding to .gitattributes for example)

Reply:
I think this is getting more and more complicated. Maybe we need something like `npm`, that packages your release. Maybe that's even something this project could do.

Not sure how .gitattributes would help with that.

Reply:
@Razzeee see for instance https://github.com/enen92/plugin.video.rtpplay/commit/1a2bd4c8825b0cb11b9edb0e666dd6cb567f0f05

The addon-submitter by romanv uses `git archive` to create the addon package before submitting automatically to the repo. If you blacklist the files you don't want to have packaged in .gitignore they won't be sent on the pull request. But yeah I agree, this somehow is getting complicated but at the same time we're working on creating a simple workflow to get everything done with little effort

Reply:
A possible option could be to merge the two projects. Also include them when using the kodi-addon-generator

Reply:
Indeed @Razzeee: if you configure the .gitattributes correctly (https://github.com/retrospect-addon/plugin.video.retrospect/blob/master/.gitattributes) then the `git archive` and the GitHub `Download zip` basically generate your release. It really saves a lot of time.

Reply:
I'll need to read up on `git archive` thank you

Reply:
I updated the code to only warn for Krypton and lower, and error for Leia and up.